ledger/pulsemanager/pulsemanager.go

```diff
@@ -397,14 +332,9 @@ func (m *PulseManager) getExecutorHotData(
 	}
 
 	requestCount := 0
-	for objID, requests := range pendingStorage.GetRequests() {
-		for reqID := range requests {
-			if _, ok := pendingRequests[objID]; !ok {
-				pendingRequests[objID] = map[core.RecordID]struct{}{}
-			}
-			pendingRequests[objID][reqID] = struct{}{}
-		}
-		requestCount += len(requests)
+	for objID, objContext := range pendingStorage.GetRequests() {
+		pendingRequests[objID] = &objContext
```

Comment:
This is not required. In fact, we don't avoid allocation here. Map elements are not addressable.

Reply:
https://goplay.space/#89tuXHapN3g